Assignment4/davidhernandog_D21C027/task07.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1E55W-V-zPge7Bj-wF55tmypQXLeOlU1w
+
+**Task 07: Querying RDF(s)**
+"""
+
+#!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+vcard = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+#OPTION1: RDFLib
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s)
+for x,y,z in g.triples((None, RDFS.subClassOf, s)): 
+  print(x)
```

Comment:
You are listing all subclasses of person and then only the subclasses of the first subclass